pkg/api/types.go

```diff
@@ -107,6 +108,29 @@ type AgentPoolProfile struct {
 	Role         AgentPoolProfileRole `json:"role,omitempty"`
 }
 
+type AuthProfile struct {
+	IdentityProviders []IdentityProvider `json:"identityProviders,omitempty"`
+}
+
+// IdentityProvider is heavily cut down equivalent to IdentityProvider in the
+// OpenShift master config.  We can add fields to match as we need to.
+type IdentityProvider struct {
+	Name     string                 `json:"name,omitempty"`
+	Provider IdentityProviderConfig `json:"provider,omityempty"`
```

Comment:
upstream on this accepts a `runtime.Object` so that different types can be swapped in.  We don't have that kind of machinery built in but is it worth considering the future of this element now?  Let's say we add in the github provider next month, what does this struct look like?

Reply:
+1 for runtime.Object. Will see if I have time to try it here. Would I keep this only in the internal model I assume?

Reply:
The external is a runtime.RawExtension in OpenShift.  Not sure following suit is strictly required as of right now (we're not guaranteeing no api changes in private preview) but it's at least something to think about.

Reply:
This is what Im testing/fighting code generation now. 
```
// IdentityProvider is heavily cut down equivalent to IdentityProvider in the
// OpenShift master config.  We can add fields to match as we need to.
type IdentityProvider struct {
	Name     string         `json:"name,omitempty"`
	Provider runtime.Object `json:"provider,omityempty"`
}

// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
// +k8s:deepcopy-gen=true
// AADIdentityProvider asdasd
type AADIdentityProvider struct {
	metav1.TypeMeta
	ClientID string `json:"clientID,omitempty"`
	Secret   string `json:"secret,omitempty"`
}
```